pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -497,7 +497,9 @@ func monitorFor(p *Propagator, clientPool dynamic.ClientPool, resource unversion
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
 				setObjectTypeMeta(newObj)
-				setObjectTypeMeta(oldObj)
+				// oldObj is obtained from the informer's cache, so it should
+				// have typeMeta set. Setting it again here causes data race.
+				// setObjectTypeMeta(oldObj)
```

Comment:
@gmarek could you be more specific? Is it the `setObjectTypeMeta` looking sketchy or not setting typemeta for oldObj sketchy? For the former, I've convince @lavalamp there is no better way in https://github.com/kubernetes/kubernetes/pull/28480#discussion_r73604648. For the latter, the oldObj is not even used by the GC, so it's safe.